apps/emqx_conf/src/emqx_conf.erl

```diff
@@ -151,6 +151,9 @@ reset(Node, KeyPath, Opts) ->
 %% @doc Called from build script.
 %% TODO: move to a external escript after all refactoring is done
 dump_schema(Dir, SchemaModule) ->
+    %% TODO: Load all apps instead of only emqx_dashboard
+    %% as this will help schemas that searches for apps with
+    %% relevant schema definitions
```

Comment:
Q: would do the injecting in [`emqx_conf_schema:roots/0`](https://github.com/emqx/emqx/blob/5aeb1feada4afedc15cacf21fb68c2495a4aedec/apps/emqx_conf/src/emqx_conf_schema.erl#L96) like auth does be a possible solution?

Reply:
I might misunderstand how that functionality works but it seems to me that it hard codes the schema module names in the emqx_conf app which I think would be good to avoid.

Reply:
For actions we currently (with this PR) hard code the info module names implicitly via `emqx_action_info:hard_coded_action_info_modules/0` which I think is a little bit better since it is in a lower level in the dependency hierarchy.